packages/@lwc/synthetic-shadow/src/polyfills/mutation-observer/polyfill.ts

```diff
@@ -72,6 +76,21 @@ function retargetMutationRecord(originalRecord: MutationRecord): MutationRecord
     return retargetedRecord;
 }
 
+// Descendant selector with a * is as performant as a selector with tag name
+// https://jsperf.com/matches-selector-optimization/1
+const ShadowedElementSelector = `[${HostMarkerAttribute}] *`;
+/**
+ * Is the given node an element and not inside a shadow?
+ * @param node
+ */
+function isNonShadowedElement(node: Node): boolean {
+    // If the target is an element and it is not the descendant of a host element
+    if (isAnElement(node) && isFalse(matches.call(node as Element, ShadowedElementSelector))) {
```

Comment:
this could be a one liner! the condition is also very weird... if the thing is not an element it returns false, if it is an element but not inside a shadow... it is true... but otherwise false... it is weird, just saying.

Reply:
Yes, it is. Initially I named it isElementShadowed(). When you pass an text/comment node in a shadow, what should the method do then, return false or true? 
So I went for the name which can certainly say the given node is not in a shadow. Everything else goes in slow path.

Reply:
I still think this should be a return statement rather than an if + return.

as for notes... this is a very interesting question, I was on the assumption that because the only place were this is called, is from the mutation observer that only observe children, by calling this with the target of the record, we should be fine:

https://dom.spec.whatwg.org/#mutationrecord

> record . target
> Returns the node the mutation affected, depending on the type. For "attributes", it is the element whose attribute changed. For "characterData", it is the CharacterData node. For "childList", it is the node whose children changed.

I don't know what are the cases where `childList` mutation's target is a node and not an element. I suspect `document` is a good case... e.g.: `new MutationObserver(function () {  }).observe(document, { childList: true })`, but that's not the case... we only observe elements, so, those are going to be target. I believe we are safe to say that our targets here are always an element.

Reply:
Caridy, this is the MutationObserver polyfill. So if components attach the MO to the document, the `target` can be a document.